File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        int productionPerHour = speed * 221;
        if (speed <= 4){
            return productionPerHour;
        }
        else if (speed <= 8){
            return productionPerHour * 0.90;
        }
        else if (speed <= 9){
            return productionPerHour * 0.80;
        }
        else{
            return productionPerHour * 0.77;
        }
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }
}

File: solutions/csharp/cars-assemble/2/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed == 10)
        {
            return 0.77;
        }

        else if (speed == 9)
        {
            return 0.8;
        }
        else if (speed > 4)
        {
            return 0.9;
        }
        else if (speed > 0)
        {
            return 1;
        }
        else return 0;
    }
    
    public static double ProductionRatePerHour(int speed) => SuccessRate(speed) * 221 * speed;

    public static int WorkingItemsPerMinute(int speed) => (int)(ProductionRatePerHour(speed)/60);

}
